lib/class-wp-block-patterns-registry.php

```diff
@@ -116,6 +116,45 @@ public static function get_instance() {
 	}
 }
 
+/**
+ * Registers a new pattern.
+ *
+ * @param string $pattern_name       Pattern name including namespace.
+ * @param array  $pattern_properties Array containing the properties of the pattern.
+ *
+ * @return boolean True if the pattern was registered with success and false otherwise.
+ */
+function register_block_pattern( $pattern_name, $pattern_properties ) {
+	return WP_Block_Patterns_Registry::get_instance()->register( $pattern_name, $pattern_properties );
+}
+
+/**
+ * Unregisters a pattern.
+ *
+ * @param string $pattern_name       Pattern name including namespace.
+ *
+ * @return boolean True if the pattern was unregistered with success and false otherwise.
+ */
+function unregister_block_pattern( $pattern_name ) {
+	return WP_Block_Patterns_Registry::get_instance()->unregister( $pattern_name );
+}
+
+/**
+ * Class used for interacting with patterns.
+ */
+class WP_Patterns_Registry {
+
+	/**
+	 * Utility method to retrieve the main instance of the class.
+	 *
+	 * @return WP_Block_Patterns_Registry The main instance.
+	 */
+	public static function get_instance() {
+		_deprecated_function( 'WP_Patterns_Registry', 'Gutenberg 8.3.0', 'WP_Block_Patterns_Registry' );
```

Comment:
I missed two things:

- Should this be 8.1.0 ? (The version at which it's deprecated, not the version at which it's removed?)
- We should add a section for the removal in 8.3.0 in [the deprecations document](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/backward-compatibility/deprecations.md)